cmd/relayer/main.go

```diff
@@ -0,0 +1,426 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/cockroachdb/apd/v3"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/urfave/cli/v2"
+
+	p2pnet "github.com/athanorlabs/go-p2p-net"
+	rcommon "github.com/athanorlabs/go-relayer/common"
+	rcontracts "github.com/athanorlabs/go-relayer/impls/gsnforwarder"
+	net "github.com/athanorlabs/go-relayer/net"
+	"github.com/athanorlabs/go-relayer/relayer"
+	rrpc "github.com/athanorlabs/go-relayer/rpc"
+
+	"github.com/athanorlabs/atomic-swap/cliutil"
+	"github.com/athanorlabs/atomic-swap/common"
+	contracts "github.com/athanorlabs/atomic-swap/ethereum"
+	swapnet "github.com/athanorlabs/atomic-swap/net"
+
+	logging "github.com/ipfs/go-log"
+)
+
+const (
+	flagDataDir          = "data-dir"
+	flagEndpoint         = "endpoint"
+	flagForwarderAddress = "forwarder-address"
+	flagKey              = "key"
+	flagRPC              = "rpc"
+	flagRPCPort          = "rpc-port"
+	flagDeploy           = "deploy"
+	flagLog              = "log-level"
+	// TODO: do we need this, or can we assume all swap relayers will need to be on the p2p network?
+	flagWithNetwork       = "with-network"
+	flagLibp2pKey         = "libp2p-key"
+	flagLibp2pPort        = "libp2p-port"
+	flagBootnodes         = "bootnodes"
+	flagRelayerCommission = "relayer-commission"
+
+	defaultLibp2pPort = 10900
+)
+
+var (
+	log = logging.Logger("main")
+
+	flags = []cli.Flag{
+		&cli.StringFlag{
+			Name:  flagDataDir,
+			Usage: "Path to store swap artifacts",
+			Value: "{HOME}/.atomicswap/{ENV}", // For --help only, actual default replaces variables
+		},
+		&cli.StringFlag{
+			Name:  flagEndpoint,
+			Value: "http://localhost:8545",
+			Usage: "Ethereum RPC endpoint",
+		},
+		&cli.StringFlag{
+			Name:  flagKey,
+			Value: "{HOME}/.atomicswap/{ENV}/relayer/eth.key",
+			Usage: "Path to file containing Ethereum private key",
+		},
+		&cli.StringFlag{
+			Name:  flagForwarderAddress,
+			Usage: "Address of the forwarder contract to use. Defaults to the forwarder address in the chain's swap contract.",
+		},
+		&cli.UintFlag{
+			Name:  flagRPCPort,
+			Value: 7799,
+			Usage: "Relayer RPC server port",
+		},
+		&cli.BoolFlag{
+			Name:  flagRPC,
+			Value: false,
+			Usage: "Run the relayer HTTP-RPC server on localhost. Defaults to false",
+		},
+		&cli.BoolFlag{
+			Name:  flagDeploy,
+			Usage: "Deploy an instance of the forwarder contract",
+		},
+		&cli.StringFlag{
+			Name:  flagLog,
+			Value: "info",
+			Usage: "Set log level: one of [error|warn|info|debug]",
+		},
+		&cli.BoolFlag{
+			Name:  flagWithNetwork,
+			Value: true,
+			Usage: "Run the relayer with p2p network capabilities",
+		},
+		&cli.StringFlag{
+			Name:  flagLibp2pKey,
+			Usage: "libp2p private key",
+			Value: common.DefaultLibp2pKeyFileName,
```

Comment:
Can we put this in the data dir like swapd does? Would it be reasonable to use the same location as `swapd` by default, or do we want to keep these identifiers separate?

Reply:
I tried to put it there but it wasn't being recognized when I ran it locally (with the integration tests), is there a trick to getting `{DATA-DIR}` to resolve? I thought it would work with the script since swapd does, probably missing something obvious

Reply:
See the answer for `flagKey`. You can look at swapd's main to see how we are handling values that depend on the `--data-dir` and `--env` flags.

Reply:
figured it out lol, updated